.github/workflows/scripts/tag_check.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -euo pipefail
```

Comment:
A nice little addition I like to add to shell scripts to add set -x only if you're debugging:

```bash
DEBUG=${DEBUG:-false}

function debug() {
  echo "Debug mode enabled" >>"$GITHUB_STEP_SUMMARY"

  # Print all environment variables
  env | sort >>"$GITHUB_STEP_SUMMARY"
  echo -e "\n" >>"$GITHUB_STEP_SUMMARY"

	set -x
}

if [[ $DEBUG == 'true' ]]; then
  debug
fi
```